signature/signature-v4.go

```diff
@@ -159,7 +173,9 @@ func V4SignVerify(r *http.Request) ErrorCode {
 	var date string
 	if date = req.Header.Get(amzDate); date == "" {
 		if date = r.Header.Get(headerDate); date == "" {
-			return errMissingDateHeader
+			if date = queryf.Get(amzDate); date == "" {
```

Comment:
I'd prefer these on two lines as at first glance it appears you've shadowed `date` using a very common go idiom but you haven't as there is an `=` instead of a `:=`

```go
date = queryf.Get(amzDate)
if date == "" {
```